api/src/main/java/com/velocitypowered/api/proxy/server/QueryResponse.java

```diff
@@ -0,0 +1,318 @@
+     */
+    @NonNull
+    public String getMap() {
+        return map;
+    }
+
+    /**
+     * Get current online player count which will be used to reply to the query.
+     * @return online player count
+     */
+    public int getCurrentPlayers() {
+        return currentPlayers;
+    }
+
+    /**
+     * Get max player count which will be used to reply to the query.
+     * @return max player count
+     */
+    public int getMaxPlayers() {
+        return maxPlayers;
+    }
+
+    /**
+     * Get proxy (public facing) hostname
+     * @return proxy hostname
+     */
+    @NonNull
+    public String getProxyHost() {
+        return proxyHost;
+    }
+
+    /**
+     * Get proxy (public facing) port
+     * @return proxy port
+     */
+    public int getProxyPort() {
+        return proxyPort;
+    }
+
+    /**
+     * Get collection of players which will be used to reply to the query.
+     * @return collection of players
+     */
+    @NonNull
+    public Collection<String> getPlayers() {
+        return players;
+    }
+
+    /**
+     * Get server software (name and version) which will be used to reply to the query.
+     * @return server software
+     */
+    @NonNull
+    public String getProxyVersion() {
+        return proxyVersion;
+    }
+
+    /**
+     * Get list of plugins which will be used to reply to the query.
+     * @return collection of plugins
+     */
+    @NonNull
+    public Collection<PluginInformation> getPlugins() {
+        return plugins;
+    }
+
+
+    /**
+     * Creates a new {@link Builder} instance from data represented by this response
+     * @return {@link QueryResponse} builder
+     */
+    @NonNull
+    public Builder toBuilder() {
+        return QueryResponse.builder()
+                .hostname(getHostname())
+                .gameVersion(getGameVersion())
+                .map(getMap())
+                .currentPlayers(getCurrentPlayers())
+                .maxPlayers(getMaxPlayers())
+                .proxyHost(getProxyHost())
+                .proxyPort(getProxyPort())
+                .players(getPlayers())
+                .proxyVersion(getProxyVersion())
+                .plugins(getPlugins());
+    }
+
+    /**
+     * Creates a new {@link Builder} instance
+     * @return {@link QueryResponse} builder
+     */
+    @NonNull
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    /**
+     * A builder for {@link QueryResponse} objects.
+     */
+    public static final class Builder {
+        private String hostname;
+        private String gameVersion;
+        private String map;
+        private int currentPlayers;
+        private int maxPlayers;
+        private String proxyHost;
+        private int proxyPort;
+        private Collection<String> players = new ArrayList<>();
+        private String proxyVersion;
+        private List<PluginInformation> plugins = new ArrayList<>();
+
+        private Builder() {}
+
+        @NonNull
```

Comment:
Remove superfluous `@NonNull` annotations. This is the default in Checker Framework.